cartography/intel/aws/wafv2.py

```diff
@@ -0,0 +1,191 @@
+    for page in paginator.paginate(LoadBalancerArns=albs):
+        for lb in page['LoadBalancers']:
+            resources["ALBs"].append(lb['DNSName'])
+
+    return resources
+
+
+@timeit
+def get_waf_acls(boto3_session: boto3.Session) -> List[Dict]:
+    wafv2_client = boto3_session.client('wafv2', config=get_botocore_config())
+    elbv2_client = boto3_session.client('elbv2', config=get_botocore_config())
+
+    acls: List[Dict] = []
+    for base_acl in wafv2_client.list_web_acls(Scope='REGIONAL')['WebACLs']:
+        acl = wafv2_client.get_web_acl(Name=base_acl['Name'], Scope='REGIONAL', Id=base_acl['Id'])['WebACL']
+        acl['Resources'] = get_load_waf_resources(wafv2_client, elbv2_client, acl['ARN'])
+        acls.append(acl)
+
+    return acls
+
+
+@timeit
+def load_waf_acl_alb_resources(neo4j_session, alb_ids: List[str], acl_id, current_aws_account_id, update_tag):
+    ingest_resources = """
+    UNWIND $ALB_IDS AS _ALB_ID
+    MATCH (acl:WAFv2WebACL{id: $ACL_ID}), (alb:LoadBalancerV2{id: _ALB_ID})
+    MERGE (acl)-[r:ASSOCIATED_WITH]->(alb)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = $UPDATE_TAG
+    WITH alb
+    MATCH (aa:AWSAccount{id: $AWS_ACCOUNT_ID})
+    MERGE (aa)-[r:RESOURCE]->(alb)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = $UPDATE_TAG
+    """
+    neo4j_session.run(
+        ingest_resources,
+        ACL_ID=acl_id,
+        ALB_IDS=alb_ids,
+        UPDATE_TAG=update_tag,
+        AWS_ACCOUNT_ID=current_aws_account_id,
+    )
+
+
+@timeit
+def load_waf_acl_apigw_resources(neo4j_session, apigw_arns, acl_id, update_tag):
+    ingest_resources = """
+    UNWIND $APIGW_IDS AS _APIGW_ID
+    MATCH (acl:WAFv2WebACL{id: $WAF_ID}), (apigw:APIGatewayStage{id: _APIGW_ID})
+    MERGE (acl)-[r:ASSOCIATED_WITH]->(apigw)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = $UPDATE_TAG
+    """
+
+    # reconstruct the APIGW ID from the ARN
+    apigw_ids = []
+    for arn in apigw_arns:
+        base = arn.split(":")[5]
+        parts = base.split("/")[-3:]
+        apigw_ids.append(f"arn:aws:apigateway:::{parts[0]}/{parts[2]}")
+
+    neo4j_session.run(
+        ingest_resources,
+        WAF_ID=acl_id,
+        APIGW_IDS=apigw_ids,
+        UPDATE_TAG=update_tag,
+    )
+
+
+@timeit
+def load_waf_acl_rules(neo4j_session, rules, acl_id, update_tag):
+    ingest_rules = """
+    UNWIND $RULES AS _RULE
+    MERGE (rule:WAFv2APIRule{name: _RULE.Name})
+    ON CREATE SET rule.firstseen = timestamp()
+    SET rule.lastupdated = $UPDATE_TAG, rule.priority = _RULE.Priority
+    WITH rule, _RULE
+    MATCH (acl:WAFv2WebACL{id: $ACL_ID})
+    MERGE (acl)-[e:ENFORCE]->(rule)
+    ON CREATE SET e.firstseen = timestamp()
+    SET e.lastupdated = $UPDATE_TAG
+    WITH rule, _RULE
+    UNWIND _RULE.Statement as _STATEMENT
+    WITH rule, _STATEMENT
+    UNWIND keys(_STATEMENT) as _STATEMENT_TYPE
+    WITH rule, _STATEMENT, _STATEMENT_TYPE,
+        CASE
```

Comment:
For this special case, I think you may do a `_transform()` step in the data before loading, and this could still work well with the NodeSchema I mentioned before.